### PR TITLE
Feature/containerize docker

### LINKED ISSUE #13: Change to PostgreSQL
We want to use PostgreSQL instead of SQLite to simulate a more likely real-world project.

### LINKED ISSUE #14: Containerize entire application
Using docker compose (to have postgres db), containerize this application for ease of development + deployment.
Includes #13

### PR DESCRIPTION
**Pull request by KELLOGGS522**
Added docker to the project by containerizing the the backend, frontend and the database. Switched the database from Sqlite3 to Postgres Closes #13.

Closes #14

To start the container run the following commands in the root of the project:
1. Only the first time or after any modification to the docker files
docker-compose build
2. To start the containers
docker-compose up -d

To create a superuser:
docker-compose exec django python manage,py createsuperuser

To stop the containers:
docker-compose down

!!IMPORTANT!!
comment the following lines of entrypoint.sh after your first time running the containers to prevent flushing all the your db data everytime:
python manage.py flush --no-input
python manage.py migrate

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+version: "3.8"
+services:
+  db:
+    image: postgres:14.3-alpine
+    volumes:
+      - ./data/db:/var/lib/postgresql/data/
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postgres
```
Comment (JR-prog) [inline docker-compose.yml:10]: DB name could be more descriptive (ex: jsuite, or whatever we plan on renaming the whole project to)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+version: "3.8"
+services:
+  db:
+    image: postgres:14.3-alpine
```
Comment (JR-prog) [inline docker-compose.yml:4]: Can you also make postgres restart always? Currently when I restart my computer everything is ready to go except the db.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    volumes:
+      - ./data/db:/var/lib/postgresql/data/
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postgres
+
+  django:
+    restart: always
+    container_name: backend
```
Comment (JR-prog) [inline docker-compose.yml:14]: Container name for all 3 should be more descriptive (ex: postgres, django, react)

File: jsuite/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# pull official base image
+FROM python:3
```
Comment (JR-prog) [inline jsuite/Dockerfile:2]: Careful here, now we have the CI running on 3.9 but I think this doesn't necessarily point to 3.9 as well (3.10.5 in my container).

File: jsuite-frontend/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# pull official base image
+FROM node:16.15
```
Comment (JR-prog) [inline jsuite-frontend/Dockerfile:2]: I'm not sure if this is happening on your end but I am getting the same error as this guy. When your containers are up can you access localhost:3000/ out of the box?

https://stackoverflow.com/questions/65055617/deploying-create-react-app-on-openshift-eacces-permission-denied-open-home

**Review by JR-prog — CHANGES_REQUESTED** [review]
Very nice work! I left a few nitpicks here and there for naming but otherwise almost everything worked seemlessly out of the box. Ideally you could also leave some docs on how to use all of this (not just setup but accessing containers and stuff for people who may not be experienced with docker)

File: docker-compose.yml (hunk 1 above)
Comment (KELLOGGS522) [reply docker-compose.yml:10]: I'll leave it at that while we decide what we're renaming the project, too but i'll open an issue for it

File: jsuite-frontend/Dockerfile (hunk 5 above)
Comment (KELLOGGS522) [reply jsuite-frontend/Dockerfile:2]: This is not an issue for me. localhost:3000/ is accessible right away. Could you elaborate more on the error and what you have to do to get it to work?

Comment (JR-prog) [reply jsuite-frontend/Dockerfile:2]: image
This is what I get when trying to go straight to localhost once the containers are up

File: jsuite/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# pull official base image
+FROM python:3.9
```
Comment (JR-prog) [inline jsuite/Dockerfile:2]: can we add an editor (maybe nano cuz we're not vim overlords) to the container by default pls?
https://stackoverflow.com/questions/30853247/how-do-i-edit-a-file-after-i-shell-to-a-docker-container

Comment (KELLOGGS522) [reply jsuite/Dockerfile:2]: Why do you want an editor? All the relevant files are mounted with volumes no?

Comment (JR-prog) [reply jsuite/Dockerfile:2]: because I am blind and did not see them.

File: jsuite-frontend/Dockerfile (hunk 5 above)
Comment (KELLOGGS522) [reply jsuite-frontend/Dockerfile:2]: This should be fixed with the latest commit

**Review by JR-prog — APPROVED** [review]

**Pull request merged**